app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/tei_events/StageViewHolder.kt

```diff
@@ -39,8 +39,9 @@ internal class StageViewHolder(
         )
         binding.lastUpdatedEvent.text = eventItem.lastUpdate.toDateSpan(itemView.context)
         binding.addStageButton.visibility =
-            if (stage.repeatable() == true &&
-                eventItem.canAddNewEvent || eventItem.eventCount == 0
+            if (
+                eventItem.canAddNewEvent && (stage.repeatable() != true && eventItem.eventCount == 0 ||
+                        stage.repeatable() == true && eventItem.eventCount == 0 || stage.repeatable() == true && eventItem.eventCount > 0 && eventItem.isSelected)
```

Comment:
can you extract this comparisons to different boolean variables for better understanding? thanks :)

Reply:
on my way!

Reply:
thanks